language/move-lang/src/typing/translate.rs

```diff
@@ -37,10 +36,9 @@ pub fn program(
     let scripts = scripts(&mut context, nscripts);
 
     assert!(context.constraints.is_empty());
-    let mut errors = context.get_errors();
-    recursive_structs::modules(&mut errors, &modules);
-    infinite_instantiations::modules(&mut errors, &modules);
-    (T::Program { modules, scripts }, errors)
+    recursive_structs::modules(context.env.get_errors_mut(), &modules);
+    infinite_instantiations::modules(context.env.get_errors_mut(), &modules);
```

Comment:
I see what get_errors_mut is used for now. I think these passes should just take a CompilationEnv